kong/plugin_service_test.go

```diff
@@ -702,7 +735,10 @@ func TestFillPluginDefaults(T *testing.T) {
 				t.Errorf(err.Error())
 			}
 
-			if diff := cmp.Diff(p, tc.expected); diff != "" {
+			// the log plugins are the only plugins that use the typedefs.lua_code type in their schema
+			// https://github.com/Kong/kong/commit/9df893f6aff98cd51f27f1c27fa30fdcf13fcf48 changes a number of other
+			// fields for 3.3, so this test only checks the relevant field to avoid needing a version split
+			if diff := cmp.Diff(p.Config["custom_fields_by_lua"], tc.expected.Config["custom_fields_by_lua"]); diff != "" {
```

Comment:
I wonder if it would be better to actually do this version split. BTW how would you see this exactly? 🤔

Reply:
See what, the failure? It popped up on the unreleased Kong test on some random Dependabot PR, and the diff is basically "the defaults for this plugin changed because its schema changed".

I don't think we'd want the version split, since we'd need to maintain two copies of the test. The field the test cares about _doesn't_ change between versions, and we don't really care about the others. This case only cares about fields with this unusual type.

The original test test uses a diff across the entire output as a simple means of handling default tests without enumerating each field of interest for the individual cases. Most of the plugins have fairly static schemas, so that shortcut doesn't break across versions.

Reply:
Alright so maybe it's just me not immediately understanding the issue here.

After taking a look at it again, it's not super clear what we're trying to test here. I get the schema change in (yet to be released) 3.3 but I don't get the goal of this test.

We're checkout `p.Config["custom_fields_by_lua"]` but not the whole `p` or `p.Config` because? 

Also:

> so this test only checks the relevant field

**Why** is only that field relevant? The test name or comments do not say why it is. 

`FillPluginDefaults()` name implies that we're filling the plugin defaults, it doesn't say anything about being specific for **a field**.

The test (or the carved out piece of it) makes it hard to figure out for a reader without a context (e.g. knowing the implementation details of `FillPluginDefaults()`).

That makes sense?